pkg/deploy/reaper/reaper.go

```diff
@@ -65,6 +66,20 @@ func (reaper *DeploymentConfigReaper) Stop(namespace, name string, timeout time.
 			// Better not error out here...
 			glog.Infof("Cannot delete ReplicationController %s/%s: %v", rc.Namespace, rc.Name, err)
 		}
+
+		// Delete all deployer and hook pods
```

Comment:
How is it an overkill? You just continue on Complete and Cancelled deployments:
```go
if deployutil.IsCancelled(rc) || deployutil.IsComplete(rc) {
   continue
}
// List deployer pods and delete them
```
In that way you also delete deployers for running deployments.